cdn/src/sentry.ts

```diff
@@ -0,0 +1,134 @@
+const APP = "worker-cli-cdn";
+
+// https://docs.sentry.io/error-reporting/configuration/?platform=javascript#environment
+const ENV = 'production';
+
+// https://docs.sentry.io/enriching-error-data/context/?platform=javascript#tagging-events
+const TAGS = { app: APP };
+
+// https://docs.sentry.io/error-reporting/configuration/?platform=javascript#server-name
+const SERVER_NAME = `${APP}-${ENV}`;
+
+// Indicates the name of the SDK client
+const CLIENT_NAME = "cloudflare-workers-sentry";
+const CLIENT_VERSION = "0.0.1";
+const RETRIES = 5;
+
+// The log() function takes an Error object and the current request
+//
+// Eg, from a worker:
+//
+// addEventListener('fetch', event => {
+//   event.respondWith(async () => {
+//     try {
+//       throw new Error('Oh no!')
+//     } catch (e) {
+//       await log(e, event.request)
+//     }
+//     return new Response('Logged!')
+//   })
+// })
+export async function log(err: Error, request: Request) {
+  // don't log if no sentry information around
+  if (typeof SENTRY_PROJECT_ID === 'undefined') return;
+  const body = JSON.stringify(toSentryEvent(err, request));
+
+  for (let i = 0; i <= RETRIES; i++) {
+    const res = await fetch(
+      `https://sentry.io/api/${SENTRY_PROJECT_ID}/store/`,
+      {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          "X-Sentry-Auth": [
+            "Sentry sentry_version=7",
+            `sentry_client=${CLIENT_NAME}/${CLIENT_VERSION}`,
+            `sentry_key=${SENTRY_KEY}`,
+          ].join(", "),
+        },
+        body,
+      }
+    );
+    if (res.status === 200) {
```

Comment:
Are we sure they don't return [`202 Accepted`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202) here when they're under higher load?  Just curious if `response.ok` is better.